staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -353,40 +363,62 @@ func (r *roundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
 			Code:   int32(res.StatusCode),
 		}
 		if err := r.a.maybeRefreshCreds(creds, resp); err != nil {
-			klog.Errorf("refreshing credentials: %v", err)
+			klog.ErrorS(err, "refreshing credentials")
 		}
 	}
 	return res, nil
 }
 
-func (a *Authenticator) credsExpired() bool {
+// contextKey type is unexported to prevent collisions.
+type contextKey int
+
+const (
+	roundTripKey contextKey = iota
+	certKey
+)
+
+func (a *Authenticator) credsExpiredLocked() bool {
 	if a.exp.IsZero() {
 		return false
 	}
 	return a.now().After(a.exp)
 }
 
-func (a *Authenticator) cert() (*tls.Certificate, error) {
-	creds, err := a.getCreds()
+func (a *Authenticator) cert(ctx context.Context) (*tls.Certificate, error) {
```

Comment:
If the change to this method is reverted, the new `TestTLSCredentialsCallsAndRotation` test reliably fails:

```
=== RUN   TestTLSCredentialsCallsAndRotation
=== CONT  TestTLSCredentialsCallsAndRotation
    exec_test.go:1566: unexpected exec call count: want=2, got=4
--- FAIL: TestTLSCredentialsCallsAndRotation (0.05s)
=== RUN   TestTLSCredentialsCallsAndRotation/valid_TLS_cert
I1203 13:37:56.814419   35159 exec.go:506] "credential was marked expired when issued" expirationTimestamp="2021-12-03 10:37:56 -0500 EST"
I1203 13:37:56.828111   35159 exec.go:506] "credential was marked expired when issued" expirationTimestamp="2021-12-03 10:37:56 -0500 EST"
2021/12/03 13:37:56 http: TLS handshake error from 127.0.0.1:49292: EOF
    exec_test.go:1550: got client.Get error: "write tcp 127.0.0.1:49292->127.0.0.1:49291: use of closed network connection", want nil
    --- FAIL: TestTLSCredentialsCallsAndRotation/valid_TLS_cert (0.02s)

=== RUN   TestTLSCredentialsCallsAndRotation/valid_TLS_cert_again
I1203 13:37:56.838715   35159 exec.go:506] "credential was marked expired when issued" expirationTimestamp="2021-12-03 10:37:56 -0500 EST"
I1203 13:37:56.852494   35159 exec.go:506] "credential was marked expired when issued" expirationTimestamp="2021-12-03 10:37:56 -0500 EST"
2021/12/03 13:37:56 http: TLS handshake error from 127.0.0.1:49293: EOF
    exec_test.go:1550: got client.Get error: "write tcp 127.0.0.1:49293->127.0.0.1:49291: use of closed network connection", want nil
    --- FAIL: TestTLSCredentialsCallsAndRotation/valid_TLS_cert_again (0.02s)
```